src/Shape.cpp

```diff
@@ -22,6 +22,78 @@ bool Shape::intersect(const Ray &, float t0, float t1, double& distance) const {
     return false;
 }
 
+
+static double side_operator(Vector3& p_U, Vector3& p_V, Vector3& q_U, Vector3& q_V){
```

Comment:
```suggestion
static double side_operator(Vector3& p_U, Vector3& p_V, Vector3& q_U, Vector3& q_V) {
```
Also needs documentation that doesn't use 'side-operator' or 'plucker coordinates' as starting points.

Your current explanation is cyclic, as in "function X implements X." No doubt it is, but unless you know what X is, that sentence has no meaning, it's fake-documentation. Try to describe what the function does without going technical, enough for a reader to decide whether the function does what it is supposed to do. (That is, is it is cause of  a problem or does it do its job and the problem must be solved elsewhere?)